index/index_test.go

```diff
@@ -412,3 +412,17 @@ func TestReaderWithInvalidBuffer(t *testing.T) {
 	_, err := NewReader(b)
 	testutil.NotOk(t, err)
 }
+
+func TestNewFileReaderCorrupted(t *testing.T) {
```

Comment:
```
// TestNewFileReaderErrorNoOpenFiles ensures that in case of an error no file remains open.
func TestNewFileReaderErrorNoOpenFiles(t *testing.T) {
```

Maybe that is a bit better. 

I think that is the last one and merging.